Tests/BSONTests/JSONTests.swift

```diff
@@ -0,0 +1,72 @@
+@testable import BSON
+import Foundation
+import Nimble
+import NIO
+import XCTest
+
+open class JSONTestCase: XCTestCase {
+    let encoder = JSONEncoder()
+    let decoder = JSONDecoder()
+
+    func testInteger() throws {
+        // Initializing a JSON with an int works, but it will be cast to a double.
+        let intJSON: JSON = 12
+        let encoded = try encoder.encode([intJSON])
+        /* JSONEncoder currently cannot encode non-object/array top level values.
+         To get around this, the generated JSON will need to be wrapped in an array
+         and unwrapped again at the end as a workaround.
+         This workaround can be removed when Swift 5.3 is the minimum supported version by the BSON library. */
+        expect(Double(String(data: encoded.dropFirst().dropLast(), encoding: .utf8)!)!)
+            .to(beCloseTo(12))
```

Comment:
ah! `beCloseTo` is a nice find! I needed something like this in a recent PR but didn't see this existed so I rounded and then asserted: 
https://github.com/mongodb/mongo-swift-driver/pull/508/files#diff-e91729b2f3ded39497c56b4c18163e40R348

I'll have to switch over to using this.